src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionCache.java

```diff
@@ -29,11 +31,14 @@
 import com.google.protobuf.Message;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 /** A {@link RemoteCache} with additional functionality needed for remote execution. */
 public class RemoteExecutionCache extends RemoteCache {
+  private final ConcurrentHashMap<Digest, ListenableFuture<Void>> uploadFutures = new ConcurrentHashMap<>();
```

Comment:
I think this can be removed.

Reply:
Maybe I shouldn't be sending Bazel patches in the middle of the night. :-/